src/ess/spectroscopy/indirect/normalisation.py

```diff
@@ -30,6 +39,51 @@ def incident_monitor_normalization(
     return lookup(monitor, dim=coords[0])[slowness]
 
 
+def monitor_position(file: NeXusFileName, monitor: MonitorName) -> MonitorPosition:
+    """Extract the position of the named monitor from a NeXus file"""
+    from scippnexus import File, compute_positions
+
+    with File(file) as data:
+        return compute_positions(data['entry/instrument'][monitor][...])['position']
+
+
+def source_monitor_path_length(
+    file: NeXusFileName, source: SourcePosition, monitor: MonitorPosition
+) -> SourceMonitorPathLength:
+    """Compute the primary spectrometer path length from source to monitor positions
+
+    Note:
+        This *requires* that the instrument group *is sorted* along the beam path.
+        HDF5 group entries are sorted alphabetically, so you should ensure that
+        the NeXus file was constructed with this in mind.
```

Comment:
For now at least we can guarantee that files conform to this requirement because I've written the NeXus Structure JSON files ECDC is using.

The Euclidean distance differs by about a millimeter on BIFROST, but could be more on other instruments.
This is arguably insignificant, but if we _can_ account for it I think we should.